spec/app/domain/authentication/authn-jwt/input_validation/parse_mandatory_claims_spec.rb

```diff
@@ -0,0 +1,253 @@
+      end
+
+      context "when input with commas at end value" do
+        subject do
+          ::Authentication::AuthnJwt::InputValidation::ParseMandatoryClaims.new().call(
+            mandatory_claims_secret_value: "claim1, claim2,"
+          )
+        end
+
+        it "raises an error" do
+          expect { subject }.to raise_error(Errors::Authentication::AuthnJwt::InvalidMandatoryClaimsFormat)
+        end
+      end
+    end
+
+    context "with empty values" do
+      context "when input with multiple connected commas value" do
+        subject do
+          ::Authentication::AuthnJwt::InputValidation::ParseMandatoryClaims.new().call(
+            mandatory_claims_secret_value: "claim1,, claim2"
+          )
+        end
+
+        it "raises an error" do
+          expect { subject }.to raise_error(Errors::Authentication::AuthnJwt::FailedToValidateClaimMissingClaimName)
+        end
+      end
+
+      context "when input with multiple connected commas with spaces value" do
+        subject do
+          ::Authentication::AuthnJwt::InputValidation::ParseMandatoryClaims.new().call(
+            mandatory_claims_secret_value: "claim1,   , claim2"
+          )
+        end
+
+        it "raises an error" do
+          expect { subject }.to raise_error(Errors::Authentication::AuthnJwt::FailedToValidateClaimMissingClaimName)
+        end
+      end
+    end
+
+    context "with claims duplications values" do
+      context "when input with connected duplicate claims value" do
+        subject do
+          ::Authentication::AuthnJwt::InputValidation::ParseMandatoryClaims.new().call(
+            mandatory_claims_secret_value: "claim1, claim2,claim2, claim3"
+          )
+        end
+
+        it "raises an error" do
+          expect { subject }.to raise_error(Errors::Authentication::AuthnJwt::InvalidMandatoryClaimsFormatContainsDuplication)
+        end
+      end
+
+      context "when input with duplicate claims value at the start and at the end" do
+        subject do
+          ::Authentication::AuthnJwt::InputValidation::ParseMandatoryClaims.new().call(
+            mandatory_claims_secret_value: "claim1, claim2,claim3, claim1"
+          )
+        end
+
+        it "raises an error" do
+          expect { subject }.to raise_error(Errors::Authentication::AuthnJwt::InvalidMandatoryClaimsFormatContainsDuplication)
+        end
+      end
+    end
+  end
+
+  context "Valid format" do
+    context "when input with 1 claim name" do
+      subject do
+        ::Authentication::AuthnJwt::InputValidation::ParseMandatoryClaims.new().call(
+          mandatory_claims_secret_value: "claim1"
+        )
+      end
+
+      it "returns a valid claims list" do
+        expect(subject).to eql(["claim1"])
+      end
+    end
+
+    context "when input with 2 words claim name" do
+      subject do
+        ::Authentication::AuthnJwt::InputValidation::ParseMandatoryClaims.new().call(
+          mandatory_claims_secret_value: "claim      1"
```

Comment:
@hilagross @sashaCher WDYT in the scenario? since i want to support this input `claim1, claim2, claim3` i delete all space from the input but as result of this logic i got the following use case which looks strange

Reply:
i can solve it but the question is what do we want to write in documentation that we ignore space? to be more strict and not allow space at all ? to solve this edge use case and allow spaces ?

Reply:
@nessiLahav do you have an example for such scenario?

Reply:
I pushed a fix please review the UT file, it contains all the possible inputs to the variable `mandatory-claims`